# Generated by experiments/misalignment-moves.py
# Apply with: cd verification/l4v && git apply <patch>

diff --git a/proof/crefine/ARM/IsolatedThreadAction.thy b/proof/refine/ARM/IsolatedThreadAction.thy
similarity index 100%
rename from proof/crefine/ARM/IsolatedThreadAction.thy
rename to proof/refine/ARM/IsolatedThreadAction.thy

diff --git a/proof/crefine/ARM/Fastpath_Equiv.thy b/proof/refine/ARM/Fastpath_Equiv.thy
similarity index 100%
rename from proof/crefine/ARM/Fastpath_Equiv.thy
rename to proof/refine/ARM/Fastpath_Equiv.thy

diff --git a/proof/crefine/ARM/Fastpath_Defs.thy b/proof/refine/ARM/Fastpath_Defs.thy
similarity index 100%
rename from proof/crefine/ARM/Fastpath_Defs.thy
rename to proof/refine/ARM/Fastpath_Defs.thy

diff --git a/proof/crefine/ARM/ArchMove_C.thy b/proof/refine/ARM/ArchMove_C.thy
similarity index 100%
rename from proof/crefine/ARM/ArchMove_C.thy
rename to proof/refine/ARM/ArchMove_C.thy

diff --git a/proof/crefine/Move_C.thy b/proof/refine/Move_C.thy
similarity index 100%
rename from proof/crefine/Move_C.thy
rename to proof/refine/Move_C.thy

diff --git a/proof/crefine/ARM/Ipc_C.thy b/proof/crefine/ARM/Ipc_C.thy
--- a/proof/crefine/ARM/Ipc_C.thy
+++ b/proof/crefine/ARM/Ipc_C.thy
@@ -10,7 +10,7 @@
   Finalise_C
   CSpace_All
   SyscallArgs_C
-  IsolatedThreadAction
+  "Refine.IsolatedThreadAction"
 begin
 
 context begin interpretation Arch . (*FIXME: arch_split*)

diff --git a/proof/crefine/ARM/Refine_C.thy b/proof/crefine/ARM/Refine_C.thy
--- a/proof/crefine/ARM/Refine_C.thy
+++ b/proof/crefine/ARM/Refine_C.thy
@@ -7,7 +7,7 @@
 chapter "Toplevel Refinement Statement"
 
 theory Refine_C
-imports Init_C Fastpath_Equiv Fastpath_C CToCRefine
+imports Init_C "Refine.Fastpath_Equiv" Fastpath_C CToCRefine
 begin
 
 context begin interpretation Arch . (*FIXME: arch_split*)

diff --git a/proof/crefine/ARM/Fastpath_C.thy b/proof/crefine/ARM/Fastpath_C.thy
--- a/proof/crefine/ARM/Fastpath_C.thy
+++ b/proof/crefine/ARM/Fastpath_C.thy
@@ -13,7 +13,7 @@
   SyscallArgs_C
   Delete_C
   Syscall_C
-  Fastpath_Defs
+  "Refine.Fastpath_Defs"
   "CLib.MonadicRewrite_C"
 begin
 

diff --git a/proof/crefine/ARM/CLevityCatch.thy b/proof/crefine/ARM/CLevityCatch.thy
--- a/proof/crefine/ARM/CLevityCatch.thy
+++ b/proof/crefine/ARM/CLevityCatch.thy
@@ -7,7 +7,7 @@
 theory CLevityCatch
 imports
   "CBaseRefine.Include_C"
-  ArchMove_C
+  "Refine.ArchMove_C"
   "CParser.LemmaBucket_C"
   "Lib.LemmaBucket"
   Boolean_C

diff --git a/proof/ROOT b/proof/ROOT
--- a/proof/ROOT
+++ b/proof/ROOT
@@ -35,16 +35,31 @@
   directories
     "$L4V_ARCH"
   theories [condition = "REFINE_QUICK_AND_DIRTY", quick_and_dirty]
+    "Move_C"
+    "$L4V_ARCH/ArchMove_C"
+    "$L4V_ARCH/IsolatedThreadAction"
+    "$L4V_ARCH/Fastpath_Defs"
+    "$L4V_ARCH/Fastpath_Equiv"
     "$L4V_ARCH/Refine"
     "$L4V_ARCH/RAB_FN"
     "$L4V_ARCH/EmptyFail_H"
     "$L4V_ARCH/Init_R"
   theories [condition = "SKIP_REFINE_PROOFS", quick_and_dirty, skip_proofs]
+    "Move_C"
+    "$L4V_ARCH/ArchMove_C"
+    "$L4V_ARCH/IsolatedThreadAction"
+    "$L4V_ARCH/Fastpath_Defs"
+    "$L4V_ARCH/Fastpath_Equiv"
     "$L4V_ARCH/Refine"
     "$L4V_ARCH/RAB_FN"
     "$L4V_ARCH/EmptyFail_H"
     "$L4V_ARCH/Init_R"
   theories
+    "Move_C"
+    "$L4V_ARCH/ArchMove_C"
+    "$L4V_ARCH/IsolatedThreadAction"
+    "$L4V_ARCH/Fastpath_Defs"
+    "$L4V_ARCH/Fastpath_Equiv"
     "$L4V_ARCH/Refine"
     "$L4V_ARCH/RAB_FN"
     "$L4V_ARCH/EmptyFail_H"
